source/Camera/InfoUtilOther/ZividBenchmark/ZividBenchmark.cpp

```diff
@@ -0,0 +1,473 @@
+        std::vector<Duration> totalDurations;
+        std::vector<std::vector<Duration>> allDurations;
+
+        for(size_t i = 0; i < numFrames; i++)
+        {
+            const auto beforeCapture = HighResClock::now();
+            const auto frame = Zivid::HDR::capture(camera, settingsVector);
+            const auto afterCapture = HighResClock::now();
+            frame.getPointCloud();
+            const auto afterProcess = HighResClock::now();
+
+            captureDurations.push_back(afterCapture - beforeCapture);
+            processDurations.push_back(afterProcess - afterCapture);
+            totalDurations.push_back(afterProcess - beforeCapture);
+        }
+        allDurations.push_back(captureDurations);
+        allDurations.push_back(processDurations);
+        allDurations.push_back(totalDurations);
+
+        printCapture3DResults(allDurations);
+
+        return totalDurations;
+    }
+
+    void benchmarkFilterProcessing(const std::vector<Duration> &captureDuration,
+                                   const std::vector<Duration> &captureDurationGaussian,
+                                   const std::vector<Duration> &captureDurationReflection,
+                                   const std::vector<Duration> &captureDurationGaussianReflection)
+    {
+        std::vector<Duration> filterProcessingDurations;
+
+        filterProcessingDurations.push_back(computeMedianDuration(captureDurationGaussian)
+                                            - computeMedianDuration(captureDuration));
+        filterProcessingDurations.push_back(computeAverageDuration(captureDurationGaussian)
+                                            - computeAverageDuration(captureDuration));
+        filterProcessingDurations.push_back(computeMedianDuration(captureDurationReflection)
+                                            - computeMedianDuration(captureDuration));
+        filterProcessingDurations.push_back(computeAverageDuration(captureDurationReflection)
+                                            - computeAverageDuration(captureDuration));
+        filterProcessingDurations.push_back(computeMedianDuration(captureDurationGaussianReflection)
+                                            - computeMedianDuration(captureDuration));
+        filterProcessingDurations.push_back(computeAverageDuration(captureDurationGaussianReflection)
+                                            - computeAverageDuration(captureDuration));
+        printFilterResults(filterProcessingDurations);
+    }
+
+    Zivid::Settings2D makeSettings2D(const size_t exposureTime)
+    {
+        Zivid::Settings2D settings;
+        settings.set(Zivid::Settings2D::ExposureTime(std::chrono::microseconds{ exposureTime }));
+        return settings;
+    }
+
+    void benchmarkCapture2D(Zivid::Camera &camera, const Zivid::Settings2D &settings, const size_t numFrames)
+    {
+        printCapture2DHeader(numFrames, settings);
+
+        for(size_t i = 0; i < 5; i++) // Warmup frames
```

Comment:
Do we mention anywhere why 5 is a good number?

Reply:
No, it is random.  The problem is that 1 was not sufficient and 10 seemed too much. I will test 1 and 2 a bit more.

Reply:
We should really get RnD involved here. @zivid/embedded-engineering

Reply:
![image](https://user-images.githubusercontent.com/46567886/77796090-f6f22800-706e-11ea-8425-6527e183d099.png)

Reply:
The two collums are 3Dacquisition times and Point cloud processing times.
This test was done with a single warmup frame.
As you can see, the second frame (warmup frame is not measured) took 136 ms, but there was another one, in the middle, that took around 250 ms.

Reply:
This could also be my low end integrated GPU (UHD 620)